### DISCUSSION TITLE
skip_dir to skip directories containing specific folders

### DISCUSSION THREAD

**Discussion by altepo** [abraunegg/onedrive#2962] [Q&A] [answered]
Thanks a lot for this onedrive client, it's great!

I would like to skip all directories that contain a .git or a .venv folder. I modified the `skip_dir` line in by config file to:

```bash
skip_dir="*/.git|*/.venv"
```

After making this change, I performed a resync. However, during testing with dummy directories containing a .git folder, I noticed the parent directory still gets uploaded. The .git folder itself is skipped as expected, thanks to `skip_dotfiles = true`, but my goal is to skip the parent directory entirely as well. I understand I could achieve manually with sync_list, but I’d prefer to avoid that if possible.

Is what I try to achieve possible? Apologies if I am making a silly mistake or missing something in the documentation.

I am running onedrive v2.4.25 on Ubuntu 24.04.

**Comment by abraunegg** [comment, marked as answer]
@altepo 
> I am running onedrive v2.4.25 on Ubuntu 24.04.

This is your problem.

Please read: https://github.com/abraunegg/onedrive?tab=readme-ov-file#supported-application-version

Then read: https://github.com/abraunegg/onedrive/blob/master/docs/ubuntu-package-install.md

Then read: 
* https://github.com/abraunegg/onedrive/blob/master/docs/usage.md#using-client-side-filtering-rules-to-determine-what-should-be-synced-with-microsoft-onedrive
* https://github.com/abraunegg/onedrive/blob/master/docs/application-config-options.md#skip_dir
